src/core/Microsoft.PowerToys.Settings.UI.Lib/GeneralSettingsCache`1.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.PowerToys.Settings.UI.Lib.Interface;
+
+namespace Microsoft.PowerToys.Settings.UI.Lib
+{
+    // This Singleton class is a wrapper around the Common settings configurations that are to be shared across all the viewmodels.
+    // This class can have only one instance and therefore the settings configurations are common to all.
+    public class GeneralSettingsCache<T> : IGeneralSettingsCache<T>
```

Comment:
I think we should rethink this name.  I believe the intent of this class is eventually provide an interface to facilitate saving and loading without forcing the view models to care about where the data is persisting as well as any implementation deatails about if the file exists, or the fact that its invoked over IPC.  SettingCache<T> or SettingRepository<T> come to mind.  IMO this file could be used for any setting file, not just 'General' Settings

Reply:
Sounds good. Yes, you're right, it is a more generic class. Shall rename it to `SettingsCache`.